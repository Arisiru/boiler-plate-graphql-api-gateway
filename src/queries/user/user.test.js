const resolver = require('./resolver');
const schema = require('./schema');
const mockUsers = require('../../mock/users');
describe('[Query.user]', () => {
  it('should compile schema', async () => {
    expect(schema).toMatchSnapshot();
  });
  const mockContext = {
    dataSources: {
      userAPI: {
        getUser: jest.fn(),
      },
    },
  };

  const { getUser } = mockContext.dataSources.userAPI;
  getUser.mockReturnValue(mockUsers.slice(0, 1));
  it('should forward mock with no changes', async () => {
    const res = await resolver.user(null, {}, mockContext);
    expect(res).toEqual(mockUsers.slice(0, 1));
  });

  it('should call corresponding data source with query arguments engagementId', async () => {
    const mockParameters = {
      email: 'joel@test.com',
    };

    await resolver.user(null, mockParameters, mockContext);
    expect(getUser).toBeCalled();
    expect(getUser).toBeCalledWith(mockParameters.email);
  });
});
