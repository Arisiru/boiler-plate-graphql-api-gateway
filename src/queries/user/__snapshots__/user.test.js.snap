// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Query.user] should compile schema 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "fields": Array [
        Object {
          "arguments": Array [
            Object {
              "defaultValue": undefined,
              "description": undefined,
              "directives": Array [],
              "kind": "InputValueDefinition",
              "name": Object {
                "kind": "Name",
                "value": "email",
              },
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          ],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "user",
          },
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "User",
              },
            },
          },
        },
      ],
      "interfaces": Array [],
      "kind": "ObjectTypeExtension",
      "name": Object {
        "kind": "Name",
        "value": "Query",
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 58,
    "start": 0,
  },
}
`;
